### PR TITLE
Add button type pl

### PR DESCRIPTION
**Pull request by karenkitchens**

### CODE REVIEW

**Comment by karenkitchens** [comment]
Closes #420

**Review by dcmouyard — APPROVED** [review]
The changes look good! 👍 

@karenkitchens Why did you add the `type="submit"` attribute to the PL demo buttons? It doesn’t affect how the buttons look, so I’m just curious if there’s something I’m overlooking.

File: source/_patterns/04-components/button/button.twig [hunk 1]
```diff
@@ -4,12 +4,12 @@
 ]|join(' ')|trim %}
 
 {% if is_button_tag %}
-  <button class="{{ classes }}"{% if is_disabled %} disabled{% endif %}>{{ text }}</button>
+  <button class="{{ classes }}"{% if is_disabled %} disabled{% endif %}{% if button_type %} type="{{button_type }}"{% endif %}>{{ text }}</button>
```
Comment (dcmouyard) [inline source/_patterns/04-components/button/button.twig]: Minor nitpick: `{{button_type }}` should be `{{ button_type }}`.

**Review by coreylafferty — APPROVED** [review]

**Comment by dcmouyard** [comment]
Fixes #342

**Pull request merged**